### PR TITLE
fix(minor-nexus-gateway)!: make msg id consistent with evm chains

### PR DESCRIPTION
**Pull request by cjcobb23**
Nexus gateway was still using colon, instead of dash in message ids, and was not prefixing with 0x.

https://axelarnetwork.atlassian.net/browse/AXE-3719

### CODE REVIEW

File: packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ impl HexTxHashAndEventIndex {
     }
 }
 
-const PATTERN: &str = "^(0x[0-9a-f]{64})-(0|[1-9][0-9]*)$";
+pub const PATTERN: &str = "^(0x[0-9a-f]{64})-(0|[1-9][0-9]*)$";
```
Comment (cgorenflo) [inline packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs:25]: why is it public now?

File: contracts/nexus-gateway/src/nexus.rs [hunk 2]
```diff
@@ -68,24 +68,37 @@ impl From<connection_router_api::Message> for Message {
...
+                    event_index: u32::try_from(msg.source_tx_index)
+                        .map_err(|_| {
+                            axelar_wasm_std::msg_id::Error::EventIndexOverflow(msg_id.clone())
+                        })
+                        .change_context(ContractError::InvalidMessageId(msg_id))?,
+                }
+                .to_string()
                 .try_into()
-                .expect("cannot be empty"),
+                .expect("must serialize message id to non-empty string"),
```
Comment (cgorenflo) [inline contracts/nexus-gateway/src/nexus.rs:95]: seems weird that this is a panic. If the underlying string was a constant I would agree, but here I'd prefer just to return an error if this fails (even if it never should, but it relies on multiple other parts to implicitly work as expected). You don't save anything by panicking here, there return type is a `Result` either way

File: contracts/nexus-gateway/src/nexus.rs [hunk 3]
```diff
@@ -68,24 +68,37 @@ impl From<connection_router_api::Message> for Message {
...
             },
             source_address: msg.source_address,
             destination_chain: msg.destination_chain,
             destination_address: msg.destination_address,
             payload_hash: msg.payload_hash,
-        }
+        })
```
Comment (fish-sammy) [inline contracts/nexus-gateway/src/nexus.rs:101]:
```suggestion
        let id = format!(
            "0x{}-{}",
            msg.source_tx_id.as_ref().encode_hex::<String>(),
            msg.source_tx_index
        );
        Ok(Self {
            cc_id: CrossChainId {
                chain: msg.source_chain,
                id: id
                    .parse::<HexTxHashAndEventIndex>()
                    .change_context(ContractError::InvalidMessageId(id))?
                    .to_string()
                    .try_into()
                    .expect("must serialize message id to non-empty string"),
            },
            source_address: msg.source_address,
            destination_chain: msg.destination_chain,
            destination_address: msg.destination_address,
            payload_hash: msg.payload_hash,
        })
```
Why not just?

File: contracts/nexus-gateway/src/contract/execute.rs [hunk 4]
```diff
@@ -2,7 +2,7 @@ use cosmwasm_std::{to_binary, Addr, Response, WasmMsg};
 use error_stack::report;
 
 use crate::error::ContractError;
-use crate::nexus;
+use crate::nexus::{self};
```
Comment (fish-sammy) [inline contracts/nexus-gateway/src/contract/execute.rs:5]: Why change this here?

File: contracts/nexus-gateway/src/contract/execute.rs [hunk 5]
```diff
@@ -24,8 +24,8 @@ where
 
         let msgs: Vec<_> = msgs
             .into_iter()
-            .map(connection_router_api::Message::from)
-            .collect();
+            .map(connection_router_api::Message::try_from)
+            .collect::<Result<Vec<connection_router_api::Message>>>()?;
```
Comment (fish-sammy) [inline contracts/nexus-gateway/src/contract/execute.rs:28]:
```suggestion
            .collect::<Result<_>>()?;
```

File: packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs (hunk 1 above)
Comment (fish-sammy) [reply packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs:25]: Same question.

**Review by fish-sammy — CHANGES_REQUESTED** [review]

File: packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs (hunk 1 above)
Comment (cjcobb23) [reply packages/axelar-wasm-std/src/msg_id/tx_hash_event_index.rs:25]: oh I was using it in nexus gateway, but I guess I'm not now, so I'll revert this

File: contracts/nexus-gateway/src/nexus.rs (hunk 3 above)
Comment (cjcobb23) [reply contracts/nexus-gateway/src/nexus.rs:101]: I don't want to hardcode the string format in nexus gateway like this. That's what led to the message id being inconsistent in the first place. The `HexTxHashAndEventIndex` defines the string format, and can do all the serialization/deserialization.

Comment (cjcobb23) [reply contracts/nexus-gateway/src/nexus.rs]: I tried to clean this up and make it simpler though, lmk what you think

**Review by cgorenflo — APPROVED** [review]

**Review by fish-sammy — APPROVED** [review]

**Pull request merged**